model/error/event.go

```diff
@@ -381,3 +380,15 @@ func addStacktraceCounter(st m.Stacktrace) {
 		frameCounter.Add(int64(frames))
 	}
 }
+
+type v2Event struct {
+	*Event
+}
+
+func (e *v2Event) Transform(tctx *transform.Context) []beat.Event {
+	events := e.Event.Transform(tctx)
+	for _, e := range events {
+		utility.Add(e.Fields, "timestamp", utility.TimeAsMicros(e.Timestamp))
```

Comment:
This isn't sitting well with me, though I see how we got here.  Have you considered adding a member to `Event` to differentiate between v1 and v2 and only add this timestamp object if v2, inside of `Event.Transform()`?

Reply:
much better, thanks!

Reply:
Have you considered creating two different event attributes for the two timestamps? This would allow setting the proper timestamp in the v1/v2 decoding methods. The transformation method could then only transform what is set, without any additional knowledge.

The reason I am suggesting this is that we could keep the whole v1/v2 separation in the decode method, as we do so far, which is also the reason that so far no separate v2 transform tests have been necessary.

Reply:
I think I see your point. If i understand correctly, you're suggesting to do something like:

```diff
type Event struct {
	Name       string
	Type       string
	Start      *float64
	Duration   float64
	Context    common.MapStr
	Stacktrace m.Stacktrace

-	Timestamp     time.Time
	TransactionId string

	// new in v2
	HexId    string
	ParentId string
	TraceId  string
+	TimestampV2     time.Time

	// deprecated in v2
	Id     *int64
	Parent *int64
+	TimestampV1     time.Time
}
```

```
func V1DecodeEvent(input interface{}, err error) (transform.Transformable, error) {
	...
	e.TimestampV1 = decoder.TimeRFC3339(raw, "timestamp")
	...
}


func V2DecodeEvent(input interface{}, err error) (transform.Transformable, error) {
	...
	e.TimestampV2 = decoder.TimeEpochMicro(raw, "timestamp")
	...
}
```

For spans, it seems to me that the transform method needs to know if it is dealing with a v1 span or a v2 span. Part of the new functionality is to set the timestamp to be reqTime + start if no timestamp is given. For v1 we're just setting it to reqTime. Looking at the `span` transform method, if none of the timestamps are set (timestamp is optional in both v1 and v2) how will we know if we should apply the v2 logic or not?

Besides the problem above, what i like about this current solution is that the decode methods "decode" to a model that is shared by v1 and v2 in the sense there's one field to denote the timestamp, regardless of which format it came in.

Having two timestamp fields for the sake of deciding the final output seems less clean to me, compared to decoding the different input representations with the same semantics into the same field as it is currently. For example, if there was some functionality that required us to use the timestamp in some logic, if we have two timestamps we would need to check them both even though the semantically mean the same thing.

Reply:
Appreciate the explanation, agree that it is cleaner this way for spans.